S7.Net/PlcAsynchronous.cs

```diff
@@ -279,6 +269,20 @@ public async Task<List<DataItem>> ReadMultipleVarsAsync(List<DataItem> dataItems
             return dataItems;
         }
 
+        private static byte[] BuildReadRequestPackage(IList<DataRequestItem> dataItems)
+        {                
+            // first create the header
+            int packageSize = 19 + (dataItems.Count * 12);
+            var package = new System.IO.MemoryStream(packageSize);
+            BuildHeaderPackage(package, dataItems.Count);
+            // package.Add(0x02);  // datenart
+            foreach (var dataItem in dataItems)
+            {
+                BuildReadDataRequestPackage(package, dataItem.DataType, dataItem.DB, dataItem.StartByteAddress, dataItem.ByteLength);
+            }
+            return package.ToArray();
+        }
```

Comment:
This _should_ be in PlcHelpers.cs, although in all honesty PlcHelpers.cs shouldn't exist at all. Also, one could say it'd be good to have methods like this available for testing, in which case they'd be better off in an internal class.

Reply:
Yeah you mentioned before that it would be great to separate concerns in the PLC functions.

I already made this static since it really does not require any PLC state.

I am also open for generally adding some folders/namespaces for all internal S7 protocol related stuff.

(Current Protocol folder has things which I don't think really belong in there)

Reply:
Wait, what?! You don't like my Protocol folder!? 😉 Yeah, even that's not coherent... Although some stuff is copied / refactored form older code, that's stuff I put there. WriteMultiple actually was one of my contributions before I became maintainer... But yeah, sure, let's start separating concerns into namespaces. Feel free to apply whatever you want, but you can also email me (or comment on the relevant PR, but you might not have a PR yet) if you want to discuss something first.

Reply:
I think a S7 folder inside the Protocol folder will do nicely. I have seen that in some other S7 library ;)